taichi/aot/graph_data.h

```diff
@@ -0,0 +1,90 @@
+#pragma once
+#include <vector>
+#include <string>
+#include <unordered_map>
+#include "taichi/aot/module_data.h"
+
+namespace taichi {
+namespace lang {
+class AotModuleBuilder;
+class Ndarray;
+namespace aot {
+// Currently only scalar and ndarray are supported.
+enum ArgKind { SCALAR, NDARRAY, UNKNOWN };
+
+/*
```

Comment:
note that doxygen style is `/**`, see https://www.doxygen.nl/manual/docblocks.html